### PR TITLE
IGNITE-9387: update interface for trainers

### PR DESCRIPTION
**Pull request by avplatonov**

### CODE REVIEW

File: modules/ml/src/main/java/org/apache/ignite/ml/knn/ann/ANNClassificationTrainer.java [hunk 1]
```diff
@@ -169,12 +203,12 @@ private ProbableLabel fillProbableLabel(int centroidIdx, CentroidStat centroidSt
                         centroidStat.put(lb, 1);
                         res.centroidStat.put(centroidIdx, centroidStat);
                     } else {
-                        int cnt = centroidStat.getOrDefault(lb, 0);
```
Comment (YuriBabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/knn/ann/ANNClassificationTrainer.java]: Why we need this change?

File: modules/ml/src/main/java/org/apache/ignite/ml/clustering/kmeans/KMeansTrainer.java [hunk 2]
```diff
@@ -127,6 +141,11 @@
...
     }
 
+    /** {@inheritDoc} */
+    @Override protected boolean checkState(KMeansModel mdl) {
+        return mdl.centers().length == k && mdl.distanceMeasure().equals(distance);
+    }
+
     /**
      * Prepares the data to define new centroids on current iteration.
      *
```
Comment (YuriBabak) [inline modules/ml/src/main/java/org/apache/ignite/ml/clustering/kmeans/KMeansTrainer.java]: missed javadoc for line 303

File: modules/ml/src/main/java/org/apache/ignite/ml/knn/ann/ANNClassificationTrainer.java (hunk 1 above)
Comment (avplatonov) [reply modules/ml/src/main/java/org/apache/ignite/ml/knn/ann/ANNClassificationTrainer.java]: misscoding
I will fix

File: modules/ml/src/main/java/org/apache/ignite/ml/clustering/kmeans/KMeansTrainer.java (hunk 2 above)
Comment (avplatonov) [reply modules/ml/src/main/java/org/apache/ignite/ml/clustering/kmeans/KMeansTrainer.java]: fixed

**Pull request closed without merging**